File: Objective-C/AlertController_ObjC/Views/ViewController.m
//
//  ViewController.m
//  AlertController_ObjC
//
//  Created by Hardik Trivedi on 06/11/19.
//  Copyright © 2019 9Brainz. All rights reserved.
//

#import "ViewController.h"
#import "Utility.h"

@interface ViewController () <UITextFieldDelegate>

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
}

- (IBAction)btnShowAlertTapped:(id)sender
{
    [[self view] endEditing:YES];
    
    NSString *strTitle = [[_txtTitle text] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    NSString *strMsg = [[_txtMsg text] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    
    if ([strTitle isEqualToString:@""] || [strMsg isEqualToString:@""]) {

        [Utility shwoAlertControllerInViewController:self withTitle:[NSString stringWithFormat:@"Empty Text"] andMessage:[NSString stringWithFormat:@"Both text required for Show this alert!"] withButtons:[NSArray arrayWithObjects:[NSString stringWithFormat:@"Ok"], nil] withCompletion:nil];
    } else {
        
        [Utility shwoAlertControllerInViewController:self withTitle:strTitle andMessage:strMsg withButtons:[NSArray arrayWithObjects:[NSString stringWithFormat:@"Ok"], nil] withCompletion:^(int index) {
               
            if (index == 0) {
                   
                NSLog(@"OK button clicked");
                // do your coding stuff here
            }
        }];
    }
}

- (IBAction)btnShowWithTwoAlertTapped:(id)sender
{
    [[self view] endEditing:YES];
    
    [Utility shwoAlertControllerInViewController:self withTitle:[NSString stringWithFormat:@"Two Button"] andMessage:[NSString stringWithFormat:@"Are you sure?"] withButtons:[NSArray arrayWithObjects:[NSString stringWithFormat:@"No"], [NSString stringWithFormat:@"Yes"], nil] withCompletion:^(int index) {
        
        if (index == 0) {
            
            NSLog(@"NO button clicked");
            // do your coding stuff here
        } else if (index == 1) {
            
            NSLog(@"YES button clicked");
            // do your coding stuff here
        }
    }];
}

- (IBAction)btnShowWithThreeAlertTapped:(id)sender
{
    [[self view] endEditing:YES];
    
    [Utility shwoAlertControllerInViewController:self withTitle:[NSString stringWithFormat:@"Three Button"] andMessage:[NSString stringWithFormat:@"Love this app! Give your important Rate about it."] withButtons:[NSArray arrayWithObjects:[NSString stringWithFormat:@"Rate Now"], [NSString stringWithFormat:@"Later"], [NSString stringWithFormat:@"No, Thanks"], nil] withCompletion:^(int index) {
        
        if (index == 0) {
            
            NSLog(@"RATE NOW button clicked");
            // do your coding stuff here
        } else if (index == 1) {
            
            NSLog(@"LATER button clicked");
            // do your coding stuff here
        } else if (index == 2) {
            
            NSLog(@"NO THANKS button clicked");
            // do your coding stuff here
        }
    }];
}

- (IBAction)btnAlertWithTitleTapped:(id)sender
{
    [[self view] endEditing:YES];
    
    [Utility shwoAlertControllerInViewController:self withTitle:[NSString stringWithFormat:@"No Internet Connection"] andMessage:nil withButtons:[NSArray arrayWithObjects:[NSString stringWithFormat:@"Ok"], nil] withCompletion:nil];
}

- (IBAction)btnAlertWithMsgTapped:(id)sender
{
    [[self view] endEditing:YES];
    
    [Utility shwoAlertControllerInViewController:self withTitle:nil andMessage:[NSString stringWithFormat:@"Task complete successfully!"] withButtons:[NSArray arrayWithObjects:[NSString stringWithFormat:@"Ok"], nil] withCompletion:nil];
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}

@end

File: Swift/AlertController_Swift/Views/ViewController.swift
//
//  ViewController.swift
//  AlertController_Swift
//
//  Created by Hardik Trivedi on 06/11/19.
//  Copyright © 2019 9Brainz. All rights reserved.
//

import UIKit

class ViewController: UIViewController
{
    
    @IBOutlet weak var txtTitle: UITextField!
    @IBOutlet weak var txtMsg: UITextField!
    

    override func viewDidLoad()
    {
        super.viewDidLoad()
        
    }
    
    @IBAction func btnShowAlertTapped(_ sender: Any)
    {
        self.view.endEditing(true)
        
        let strTitle: String = (self.txtTitle.text?.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines))!
        let strMsg: String = (self.txtMsg.text?.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines))!
        
        if strTitle == "" || strMsg == "" {
            
            Utility.shwoAlertControllerInViewController(vc: self, withTitle: "Empty Text", andMessage: "Both text required for Show this alert!", withButtons: ["Ok"], completion: nil)
        } else {
            
            Utility.shwoAlertControllerInViewController(vc: self, withTitle: strTitle, andMessage: strMsg, withButtons: ["Ok"]) { (intIndex) in
                
                if intIndex == 0 {
                    
                    print("OK button clicked")
                    // do your coding stuff here
                }
            }
        }
    }
    
    @IBAction func btnShowWithTwoAlertTapped(_ sender: Any)
    {
        self.view.endEditing(true)
        
        Utility.shwoAlertControllerInViewController(vc: self, withTitle: "Two Button", andMessage: "Are you sure?", withButtons: ["No", "Yes"]) { (intIndex) in
            
            if intIndex == 0 {
                
                print("NO button clicked")
                // do your coding stuff here
            } else if intIndex == 1 {
                
                print("YES button clicked")
                // do your coding stuff here
            }
        }
    }
    
    @IBAction func btnShowWithThreeAlertTapped(_ sender: Any)
    {
        self.view.endEditing(true)
        
        Utility.shwoAlertControllerInViewController(vc: self, withTitle: "Three Button", andMessage: "Love this app! Give your important Rate about it.", withButtons: ["Rate Now", "Later", "No, Thanks"]) { (intIndex) in
            
            if intIndex == 0 {
                
                print("RATE NOW button clicked")
                // do your coding stuff here
            } else if intIndex == 1 {
                
                print("LATER button clicked")
                // do your coding stuff here
            } else if intIndex == 2 {
                           
                print("NO THANKS button clicked")
                // do your coding stuff here
            }
        }
    }
    
    @IBAction func btnAlertWithTitleTapped(_ sender: Any)
    {
        self.view.endEditing(true)
        
        Utility.shwoAlertControllerInViewController(vc: self, withTitle: "No Internet Connection", andMessage: nil, withButtons: ["Ok"], completion: nil)
    }
    
    @IBAction func btnAlertWithMsgTapped(_ sender: Any)
    {
        self.view.endEditing(true)
        
        Utility.shwoAlertControllerInViewController(vc: self, withTitle: nil, andMessage: "Task complete successfully!", withButtons: ["Ok"], completion: nil)
    }
    
}

extension ViewController: UITextFieldDelegate
{
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool
    {
        textField.resignFirstResponder()
        return true
    }
    
}
